mongodbatlas/data_source_mongodbatlas_cloud_shared_tier_restore_jobs.go

```diff
@@ -0,0 +1,135 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+// This datasource does not have a resource: we tested it manually
+func dataSourceMongoDBAtlasCloudSharedTierRestoreJobs() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMongoDBAtlasCloudSharedTierRestoreJobRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"results": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"job_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_deployment_item_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_scheduled_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"delivery_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expiration_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasCloudSharedTierRestoreJobRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).AtlasV2
+
+	projectID := d.Get("project_id").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	jobs, resp, err := conn.SharedTierRestoreJobsApi.ListSharedClusterBackupRestoreJobs(ctx, clusterName, projectID).Execute()
+	defer resp.Body.Close()
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting shared tier restore jobs for cluster '%s': %w", clusterName, err))
```

Comment:
FYI: this statement will print error in the format that is default for the SDK (which is a generic and might not be desired for terraform). For terraform we might want to override the format or check for specific error codes. 

For more info see docs:

https://github.com/mongodb/atlas-sdk-go/blob/5b4da017ab144760a7ce4af23f42d83c1844dbe6/docs/docs/SharedTierRestoreJobsApi.md?plain=1#L217C28-L217C28

And 

https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_2_error_handling.md#checking-for-the-existence-of-a-specific-error-code

Reply:
No, we don't apply any transformation to the error coming from the API. Btw, thanks for sharing this as it is going to be useful for other resources when we need to access the error code (sometime we skip the deletion if the GET resp returns 404)

Reply:
Error is formatted by the SDK. Here is the format:
https://github.com/mongodb/atlas-sdk-go/blob/main/admin/client.go#L764-L768